plugin/src/test/resources/com/google/errorprone/xplat/checker/testdata/J2objcMethodNameMangle.java

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.errorprone.xplat.checker.testdata;
+
+import com.google.j2objc.annotations.ObjectiveCName;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Set;
+
+public class J2objcMethodNameMangle {
+
+  // BUG: Diagnostic contains: ComGoogleErrorproneXplatCheckerTestdataJ2objcMethodNameMangle_hello
+  public static HashMap<Object, Set<String>> hello() {
+    return new HashMap<Object, Set<String>>();
+  }
+
+  // BUG: Diagnostic contains: ComGoogleErrorproneXplatCheckerTestdataJ2objcMethodNameMangle_helloWithJavaUtilHashMap_
+  public static HashMap<Object, Set<String>> hello(HashMap<Object, Set<String>> x) {
+    return x;
+  }
+
+  // BUG: Diagnostic contains: ComGoogleErrorproneXplatCheckerTestdataJ2objcMethodNameMangle_helloWithJavaUtilHashMap_withJavaUtilHashMap_withJavaUtilHashMap_withJavaUtilHashMap_withJavaUtilHashMap_withJavaUtilHashMap_withJavaUtilHashMap_withJavaUtilHashMap_withJavaUtilHashMap_
+  public static HashMap<Object, Set<String>> hello(HashMap<Object, Set<String>> x,
+      HashMap<Object, Set<String>> y,
+      HashMap<Object, Set<String>> z, HashMap<Object, Set<String>> q,
+      HashMap<Object, Set<String>> r,
+      HashMap<Object, Set<String>> s, HashMap<Object, Set<String>> t,
+      HashMap<Object, Set<String>> u, HashMap<Object, Set<String>> v) {
+    return x;
+  }
+
+  @ObjectiveCName("renamedMethod")
+  // BUG: Diagnostic contains: ComGoogleErrorproneXplatCheckerTestdataJ2objcMethodNameMangle_renamedMethod
+  public static void hi() {
+    return;
+  }
+
+  @ObjectiveCName("Middleclass")
+  private static class Test {
```

Comment:
Do we have a test with a method that has at least one parameter of type `Test`, and would the mangled method preview show the correct mangled name? (That is it should be `ComGoogleErrorproneXplatCheckerTestdataMiddleClass someArg` instead of `ComGoogleErrorproneXplatCheckerTestdataJ2ObjCMethodNameMangle_Test someArg`)?

Reply:
There is currently no such test, I have added some.